features/FEATURE_COMMON_PAL/nanostack-hal-mbed-cmsis-rtos/arm_hal_timer.cpp

```diff
@@ -12,15 +12,25 @@
 
 static Timer timer;
 static Timeout timeout;
+
+// High priority event queue is needed if the timer callback is not interrupt safe or the
+// critical section implementation is not disabling interrupts.
```

Comment:
Phrasing is a bit wonky above - this isn't an "or". If the critical section is not disabling interrupts, then it must be claiming a mutex, which means the timer callback is not interrupt safe, and any other code doing critical sections won't be protected against a called-from-interrupt timer.

Maybe something like

    // If critical sections are implemented using mutexes, timers must be called in thread context, and
    // we use the high-priority event queue for this.
    // If critical sections disable interrupts, we can call timers directly from interrupt. Avoiding the
    // event queue can save ~1600B of RAM if the rest of the system is not using the event queue either.
    // Caveats of this tunable are listed on arm_hal_interrupt.c.

Reply:
Your version is much better, changed to it and force pushed.